### PR TITLE
Remove sample_prior from EnKFMain and make it stand-alone

### PR DESCRIPTION
**Pull request by oyvindeide**
**Issue**
Resolves #my_issue

**Approach**
_Short description of the approach_

(Screenshot of new behavior in GUI if applicable)

## Pre review checklist

- [x] Read through the code changes carefully after finishing work
- [x] Make sure tests pass locally (after every commit!)
- [x] Prepare changes in small commits for more convenient review (optional)
- [x] PR title captures the intent of the changes, and is fitting for release notes.
- [x] Updated documentation
- [x] Ensured that unit tests are added for all new behavior (See 
    [Ground Rules](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md#ground-rules)),
    and changes to existing code have good test coverage.

## Pre merge checklist
- [x] Added appropriate release note label
- [x] Commit history is consistent and clean, in line with the [contribution guidelines](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md).

### CODE REVIEW

File: src/ert/enkf_main.py [hunk 1]
```diff
@@ -215,7 +216,7 @@ def loadFromForwardModel(
         t = time.perf_counter()
         run_context = self.ensemble_context(fs, realization, iteration)
         nr_loaded = fs.load_from_run_path(
-            self.getEnsembleSize(),
```
Comment (sondreso) [inline src/ert/enkf_main.py:218]: Are we guaranteed that ensemble size is always set in storage here?

File: tests/unit_tests/run_models/test_base_run_model.py [hunk 2]
```diff
@@ -4,12 +4,23 @@
...
+)
+
+
+@pytest.fixture
+def base_arguments():
+    return SimulationArguments(random_seed=1234)
 
 
-def test_base_run_model_supports_restart(minimum_case):
+def test_base_run_model_supports_restart(minimum_case, base_arguments):
```
Comment (sondreso) [inline tests/unit_tests/run_models/test_base_run_model.py:18]: `base_arguments` not used?

File: tests/unit_tests/test_load_forward_model.py [hunk 3]
```diff
@@ -170,7 +170,7 @@ def test_load_forward_model_summary(summary_configuration, storage, expected, ca
         responses=ert_config.ensemble_config.response_configuration
     )
     prior_ensemble = storage.create_ensemble(
-        experiment_id, name="prior", ensemble_size=100
+        experiment_id, name="prior", ensemble_size=1
```
Comment (sondreso) [inline tests/unit_tests/test_load_forward_model.py:173]: Why the change in size?

File: src/ert/enkf_main.py (hunk 1 above)
Comment (oyvindeide) [reply src/ert/enkf_main.py:218]: Yes, but will split this into another commit, or perhaps PR, dont have to do that here.

File: tests/unit_tests/test_load_forward_model.py (hunk 3 above)
Comment (oyvindeide) [reply tests/unit_tests/test_load_forward_model.py:173]: It was related to the `getEnsembleSize()` change, undid both

File: tests/unit_tests/run_models/test_base_run_model.py (hunk 2 above)
Comment (oyvindeide) [reply tests/unit_tests/run_models/test_base_run_model.py:18]: It is now 👍

**Review by sondreso — APPROVED** [review]
LGTM :ship:
